modules/pricer/src/test/java/com/opengamma/strata/pricer/rate/future/DiscountingDeliverableSwapFutureProductPricerTest.java

```diff
@@ -0,0 +1,181 @@
+  private static final double[] USD_FWD3_TIME = new double[] {0.0, 0.5, 1.0, 2.0, 5.0, 10.0 };
+  private static final double[] USD_FWD3_RATE = new double[] {0.0150, 0.0125, 0.0150, 0.0175, 0.0150, 0.0150 };
+  private static final CurveName USD_FWD3_NAME = CurveName.of("USD LIBOR 3M");
+  private static final CurveMetadata USD_FWD3_METADATA = Curves.zeroRates(USD_FWD3_NAME, ACT_ACT_ISDA);
+  private static final InterpolatedNodalCurve USD_FWD3 =
+      InterpolatedNodalCurve.of(USD_FWD3_METADATA, USD_FWD3_TIME, USD_FWD3_RATE, INTERPOLATOR);
+  private static final ImmutableRatesProvider PROVIDER = ImmutableRatesProvider.builder()
+      .valuationDate(VALUATION)
+      .fxMatrix(FxMatrix.empty())
+      .discountCurves(ImmutableMap.of(USD, USD_DSC))
+      .indexCurves(ImmutableMap.of(USD_LIBOR_3M, USD_FWD3))
+      .timeSeries(ImmutableMap.of(USD_LIBOR_3M, LocalDateDoubleTimeSeries.empty()))
+      .build();
+  // underlying swap
+  private static final NotionalSchedule UNIT_NOTIONAL = NotionalSchedule.of(USD, 1d);
+  private static final HolidayCalendar CALENDAR = HolidayCalendars.SAT_SUN;
+  private static final BusinessDayAdjustment BDA_MF = BusinessDayAdjustment.of(MODIFIED_FOLLOWING, CALENDAR);
+  private static final BusinessDayAdjustment BDA_P = BusinessDayAdjustment.of(PRECEDING, CALENDAR);
+  private static final LocalDate START = LocalDate.of(2012, 12, 19);
+  private static final LocalDate END = START.plusYears(10);
+  private static final double RATE = 0.0175;
+  private static final SwapLeg FIXED_LEG = RateCalculationSwapLeg.builder()
+      .payReceive(RECEIVE)
+      .accrualSchedule(PeriodicSchedule.builder()
+          .startDate(START)
+          .endDate(END)
+          .frequency(P6M)
+          .businessDayAdjustment(BDA_MF)
+          .stubConvention(StubConvention.SHORT_FINAL)
+          .build())
+      .paymentSchedule(PaymentSchedule.builder()
+          .paymentFrequency(P6M)
+          .paymentDateOffset(DaysAdjustment.NONE)
+          .build())
+      .notionalSchedule(UNIT_NOTIONAL)
+      .calculation(FixedRateCalculation.builder()
+          .dayCount(THIRTY_U_360)
+          .rate(ValueSchedule.of(RATE))
+          .build())
+      .build();
+  private static final SwapLeg IBOR_LEG = RateCalculationSwapLeg.builder()
+      .payReceive(PAY)
+      .accrualSchedule(PeriodicSchedule.builder()
+          .startDate(START)
+          .endDate(END)
+          .frequency(P3M)
+          .businessDayAdjustment(BDA_MF)
+          .stubConvention(StubConvention.SHORT_FINAL)
+          .build())
+      .paymentSchedule(PaymentSchedule.builder()
+          .paymentFrequency(P3M)
+          .paymentDateOffset(DaysAdjustment.NONE)
+          .build())
+      .notionalSchedule(UNIT_NOTIONAL)
+      .calculation(IborRateCalculation.builder()
+          .index(USD_LIBOR_3M)
+          .fixingDateOffset(DaysAdjustment.ofBusinessDays(-2, CALENDAR, BDA_P))
+          .build())
+      .build();
```

Comment:
It may be worth using `FixedIborSwapConventions` to create swaps for testing purposes.